tools/stress-cluster/scripts/deploy_stress_tests.ps1

```diff
@@ -0,0 +1,160 @@
+function Run() {
+    Write-Output "" "==> $args" ""
+    $command, $arguments = $args
+    & $command $arguments
+    if ($LASTEXITCODE) {
+        Write-Error "Command '$args' failed with code: $LASTEXITCODE" -ErrorAction 'Continue'
+        $FailedCommands.Add(@{ command = "$args"; code = $LASTEXITCODE })
+    }
+}
+
+function RunOrExit() {
+    run @args
+    if ($LASTEXITCODE) {
+        exit $LASTEXITCODE
+    }
+}
+
+function Login([string]$subscription, [string]$clusterGroup, [boolean]$pushImages) {
+    Write-Output "Logging in to subscription, cluster and container registry"
+    az account show
+    if ($LASTEXITCODE) {
+        RunOrExit az login --allow-no-subscriptions
+    }
+
+    $clusterName = (az aks list -g $clusterGroup -o json| ConvertFrom-Json).name
+
+    RunOrExit az aks get-credentials `
+        -n "$clusterName" `
+        -g "$clusterGroup" `
+        --subscription "$subscription" `
+        --overwrite-existing
+
+    if ($pushImages) {
+        $registry = (az acr list -g $clusterGroup -o json | ConvertFrom-Json).name
+        RunOrExit az acr login -n $registry
+    }
+}
+
+function DeployStressTests(
+    [string]$searchDirectory = '.',
+    [hashtable]$filters = @{},
+    [string]$environment = 'test',
+    [string]$repository = 'images',
+    [boolean]$pushImages = $false,
+    [string]$clusterGroup = 'rg-stress-test-cluster-',
+    [string]$deployId = 'local',
+    [string]$subscription = 'Azure SDK Test Resources'
+) {
+    if ($PSCmdlet.ParameterSetName -eq 'DoLogin') {
+        Login $subscription $clusterGroup $pushImages
+    }
+
+    RunOrExit helm repo add stress-test-charts https://stresstestcharts.blob.core.windows.net/helm/
+    Run helm repo update
+    if ($LASTEXITCODE) { return $LASTEXITCODE }
+
+    $pkgs = FindStressPackages $searchDirectory $filters
+    Write-Output "" "Found $($pkgs.Length) stress test packages:"
+    Write-Output $pkgs.Directory ""
+    foreach ($pkg in $pkgs) {
+        Write-Output "Deploying stress test at '$($pkg.Directory)'"
+        DeployStressPackage $pkg $deployId $environment $repository $pushImages
+    }
+
+    Write-Output "Releases deployed by $deployId"
+    Run helm list --all-namespaces -l deployId=$deployId
+
+    if ($FailedCommands) {
+        Write-Warning "" "The following commands failed:" ""
+        foreach ($cmd in $FailedCommands) {
+            Write-Error "'$($cmd.command)' failed with code $($cmd.code)" -ErrorAction 'Continue'
+        }
+        exit 1
+    }
+}
+
+function DeployStressPackage(
+    [object]$pkg,
+    [string]$deployId,
+    [string]$environment,
+    [string]$repository,
+    [boolean]$pushImages
+) {
+    $registry = (az acr list -g $clusterGroup -o json | ConvertFrom-Json).name
+    if (!$registry) {
+        Write-Output "Could not find container registry in resource group $clusterGroup"
+        exit 1
+    }
+
+    if ($pushImages) {
+        Run helm dependency update $pkg.Directory
```

Comment:
RunExitOnFailure?

Reply:
Oh nevermind I see these you just want to return not next and you are capturing the failures in a colleciton.